src/app/dashboard/whitelist/WhiteListPage.js

```diff
@@ -0,0 +1,378 @@
+  previewCSVShowing: state.whitelist.previewCSVShowing,
+  listLength: state.whitelist.listLength,
+})
+
+const mapDispatchToProps = (dispatch: Function) => ({
+  initialize: () => dispatch(initialize()),
+  handleUpload: (htmlCsv) => dispatch(uploadCSV(htmlCsv.target.files[0])),
+  multiSubmit: () => dispatch(multiUserSubmit()),
+  singleSubmit: () => dispatch(oneUserSubmit()),
+  getWhitelist: (calenderStart: Date, calenderEnd: Date) => dispatch(getWhitelist(calenderStart, calenderEnd)),
+  updateListLength: (pageNumber: number) => dispatch(listLength(pageNumber)),
+  removeInvestor: (investors: Array<string>) => dispatch(removeInvestor(investors)),
+  editInvestors: (investors: Array<string>) => dispatch(editInvestors(investors)),
+})
+
+type Props = StateProps & DispatchProps
+
+type State = {
+  page: number,
+  editInvestorsShowing: boolean,
+  editInvestors: Array<string>,
+}
+
+ type EventData = {
+  id: string,
+  address: string,
+  added: ?string,
+  addedBy: ?string,
+  from: string,
+  to: string,
+}
+
+type PageChanger = {
+  page: number,
+  pageSize: number,
+}
+
+type DatePickerType = [Date, Date]
+
+const dateFormat = (date: Date) => date.toLocaleDateString('en', {
+  year: 'numeric',
+  month: 'short',
+  day: 'numeric',
+})
+
+class WhitelistPage extends Component<Props, State> {
+  state = {
+    page: 0,
+    editInvestorsShowing: false,
+    editInvestors: [],
+  }
+
+  componentWillMount () {
+    this.props.initialize()
+  }
+
+  handleChangePages = (pc: PageChanger) => {
+    this.props.updateListLength(pc.pageSize)
+    this.setState({
+      page: (pc.page - 1),
+    })
+  }
+
+  handleDatePicker = (picker: DatePickerType) => {
+    if (picker.length === 2){
+      this.setState({
+        page: 0, //reset to initial page , othewise it might refer to a page that doesnt exist //TODO: make sure this is true, im not sure it is
+      })
+      this.props.getWhitelist(picker[0],picker[1])
+    }
+  }
+
+  handleEditInvestors = (investors: Array<Object>) => {
+    let addresses = []
+    for (let i = 0; i < investors.length; i++) {
+      addresses.push(investors[i].cells[0].value)
+    }
+    this.setState({
+      editInvestorsShowing: true,
+      editInvestors: addresses,
+    })
+  }
+
+  handleRequestSubmit = () => {
+    this.props.editInvestors(this.state.editInvestors)
+    this.setState({
+      editInvestorsShowing: false,
+    })
+  }
+
+  handleRequestClose = () => {
+    this.setState({
+      editInvestorsShowing: false,
+    })
+  }
+
+  onHandleInvestorSubmit = () => {
+    this.props.singleSubmit()
+    return true  //needed for the component from carbon to work properly
+  }
+
+  //renders the list by making it date strings and splitting up in pages, at the start of the render function
+  paginationRendering () {
+    let paginatedArray = []
+    let investors = this.props.investors
+    let pageNum = this.state.page
+    let listLength = this.props.listLength
+    let startSlice = pageNum*listLength
```

Comment:
missing spaces before and after operator

Reply:
fixed